signac/core/file_buffered_synced_collection.py

```diff
@@ -0,0 +1,283 @@
+# Copyright (c) 2020 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+"""Implement the FileBufferedSyncedCollection class.
+
+The FileBufferedSyncedCollection is a concrete implementation of the buffer
+protocol established by the BufferedSyncedCollection class. It uses an
+in-memory cache to store data when in buffered mode. It is suitable for
+use with any file-based backend because it performs integrity checks based on
+whether or not the underlying file has been modified while buffering was
+activated.
+"""
+
+import errno
+import hashlib
+import json
+import os
+import sys
+
+from typing import Any, Dict
+
+from .buffered_synced_collection import BufferedCollection
+from .errors import MetadataError
+
+
+def get_buffer_capacity():
+    """Get the current buffer capacity."""
+    return FileBufferedCollection._BUFFER_CAPACITY
+
+
+def set_buffer_capacity(new_capacity):
+    """Update the buffer capacity."""
+    FileBufferedCollection._BUFFER_CAPACITY = new_capacity
```

Comment:
This call should probably raise `ValueError` or some buffer-specific exception if the current buffer size is larger than the new capacity (otherwise the next store operation will fail, right?). At least, that behavior should be clearly documented.

Reply:
Good catch. Rather than raising an error, shouldn't that just trigger a flush?